Added option to enable S3 Transfer Acceleration

djdenv opened the pull request:
Adds an option in S3 to enable Transfer Acceleration. Will work only if the supplied bucket supports it. I also did a little bit of code cleanup in the classes I touched, marking fields final, etc...

DavidTanner commented on src/main/java/org/jenkinsci/plugins/awsbeanstalkpublisher/extensions/AWSEBS3Setup.java:
```diff
@@ -68,10 +69,16 @@ public String getExcludes() {
     }
 
-    private final Boolean overwriteExistingFile;
+    private final boolean overwriteExistingFile;
```
I am not sure this is safe because fields don't get initialized from storage.  This will cause issues for projects that upgrade without setting the value.

DavidTanner commented on src/main/java/org/jenkinsci/plugins/awsbeanstalkpublisher/AWSEBS3Uploader.java:
```diff
@@ -83,50 +82,68 @@ public void uploadArchive(AWSElasticBeanstalk awseb) throws Exception {
 
-    private File getLocalFileObject(FilePath rootFileObject) throws Exception {
-        File resultFile = File.createTempFile("awseb-", ".zip");
+    private File getLocalFileObject(final FilePath rootFileObject) throws Exception {
```
Why all of the final?  I would prefer not to change them all just because.

DavidTanner commented on src/main/java/org/jenkinsci/plugins/awsbeanstalkpublisher/AWSEBS3Uploader.java:
```diff
@@ -83,50 +82,68 @@ public void uploadArchive(AWSElasticBeanstalk awseb) throws Exception {
+                uploadFile = isOverwriteExistingFile;
             }
-        } catch (AmazonS3Exception s3e) {
+        }
```
Not a fan of the Microsoft style brackets.  Can you change them back please?

djdenv commented:
I added the finals because it's generally a good practice to do so. My IDE (IntelliJ) actually does it for me when I commit files. I changed the Boolean to boolean simply because I saw no reason to have a check for null values. A boolean will always be false if not set. I don't know too much about how Jenkins works with plugins though, especially for backward compatibility with older versions, so I'll defer to your judgement on these. I made those changes, so I think this should be ready for merging.
